jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -681,6 +717,32 @@ public MethodHandle findVirtual(Class<?> clazz, String methodName, MethodType ty
 			handle = SecurityFrameInjector.wrapHandleWithInjectedSecurityFrameIfRequired(this, handle);
 			return handle;
 		}
+
+		/*[IF Sidecar19-SE]*/
+		/**
+		 * Check if targetClass is in a package visible from the accessModule
+		 * @param accessModule module of the referring class
+		 * @param targetClass Class which the referring class is accessing
+		 * @param throwError Select the throwable thrown if the targetClass is not visible
+		 * @throws IllegalAccessError thrown if throwError is true
+		 * @throws IllegalAccessException thrown if throwError is false
+		 */
+		static void checkClassModuleVisibility(Module accessModule, Class<?> targetClass, boolean throwError) 
+				throws IllegalAccessError, IllegalAccessException {
+			Module targetModule = targetClass.getModule();
+			if (!Objects.equals(accessModule, targetModule)
+					&& !(targetModule.isExported(targetClass.getPackageName(), accessModule)
+							&& accessModule.canRead(targetModule))) {
+				/*[MSG "K0664", "{0} is not in an exported package"]*/
```

Comment:
Can you add the two modules to this message?  It will be easier for a user to resolve an access failure if they get the module info in the exception.